src/skadi/sample.cc

```diff
@@ -122,35 +123,76 @@ class cache_item_t {
 
   bool unpack(const char* unpacked) {
     this->unpacked = unpacked;
-    // for setting where to read compressed data from
-    auto src_func = [this](z_stream& s) -> void {
-      s.next_in = static_cast<Byte*>(static_cast<void*>(data.get()));
-      s.avail_in = static_cast<unsigned int>(data.size());
-    };
-
-    // for setting where to write the uncompressed data to
-    auto dst_func = [this](z_stream& s) -> int {
-      s.next_out = (Byte*)(this->unpacked);
-      s.avail_out = HGT_BYTES;
-      return Z_FINISH; // we know the output will hold all the input
-    };
-
-    // we have to unzip it
-    if (!baldr::inflate(src_func, dst_func)) {
-      LOG_WARN("Corrupt compressed elevation data");
+
+    if (format == format_t::GZIP) {
+      // for setting where to read compressed data from
+      auto src_func = [this](z_stream& s) -> void {
+        s.next_in = static_cast<Byte*>(static_cast<void*>(data.get()));
+        s.avail_in = static_cast<unsigned int>(data.size());
+      };
+
+      // for setting where to write the uncompressed data to
+      auto dst_func = [this](z_stream& s) -> int {
+        s.next_out = (Byte*)(this->unpacked);
+        s.avail_out = HGT_BYTES;
+        return Z_FINISH; // we know the output will hold all the input
+      };
+
+      // we have to unzip it
+      if (!baldr::inflate(src_func, dst_func)) {
+        LOG_WARN("Corrupt gzip elevation data");
+        format = format_t::UNKNOWN;
+        return false;
+      }
+    } else if (format == format_t::LZ4) {
+      LZ4F_decompressionContext_t decode;
+      LZ4F_decompressOptions_t options;
+      LZ4F_createDecompressionContext(&decode, LZ4F_VERSION);
```

Comment:
can you refresh my memory, does this also support LZ4_HC or other variants? i vaguely remember there were 3 formats, one was roughly normal but not fixed endian, another was fixed endian (probably network order) and one was "high compression". does this support all of them automatically or just one of them, and if the latter, which one? it was 3 years ago when i implemented it and i think i ended up going with the first one becuase its what the command line tool spits out maybe the others were containerless and more for streaming situations? sorry i just cant remember

Reply:
No worries, @kevinkreiser.

I just went with whatever the command line tool spit out. There's other formats that seem to be more for streaming, but it seemed the most logical target was the command line results. I made a few mistakes on the decompression side using `lz4.h` functions instead of `lz4_frame.h` functions, so it appears at least the `lz4` command line outputs a framed format?

Should we just state we support files compressed with `lz4` command line in the release notes? The command line doesn't seem to offer any options for this (https://linuxcommandlibrary.com/man/lz4).

Reply:
nope thats all good. i just wanted to confirm my recollection. ill review this tonight hopefully when i have a free moment. thanks for the efforts!